### PR TITLE
[sensor] k64f support

### PR DESCRIPTION
**Pull request by jimmy-huang**
Refactor sensor and decouple IPM logic into a separate sensor_board
module.  Added initial support for Accelerometer for the FRDM-K64F
board.

Signed-off-by: Jimmy Huang <jimmy.huang@intel.com>

### CODE REVIEW

File: src/zjs_sensor_frdm.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (grgustaf) [inline src/zjs_sensor_frdm.json]: Rename this file from frdm to k64f, that's what we've been using elsewhere as a short abbreviation.

File: src/zjs_sensor_gyro.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
     "module": "sensor_gyro",
     "constructor": "Gyroscope",
     "depends": ["sensor"],
-    "targets": ["arduino_101"],
+    "targets": ["arduino_101", "frdm_k64f"],
```
Comment (grgustaf) [inline src/zjs_sensor_gyro.json]: Not sure this makes sense if as per IRC it's not actually going to work on k64f.

File: src/zjs_sensor_board_frdm.c [hunk 3]
```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2016-2017, Intel Corporation.
```
Comment (grgustaf) [inline src/zjs_sensor_board_frdm.c]: As below use `k64f` not `frdm` in filename, too generic.

**Comment by jimmy-huang** [comment]
@grgustaf updated patch

**Comment by grgustaf** [comment]
+1

**Pull request merged**